analyses/create-subset-files/01-create_subset_files.Rmd

```diff
@@ -0,0 +1,370 @@
+```
+
+## SNV
+
+SNV files are in [annotated MAF format](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/format/vep-maf.md).
+
+These are a bit unwieldy to read in, so we'll split up the chunks below.
+
+```{r}
+mutect2 <- data.table::fread(file.path(data_dir, mutect2_file),
+                             skip = 1,  # skip version string
+                             data.table = FALSE)
+```
+
+```{r}
+strelka2 <- data.table::fread(file.path(data_dir, strelka2_file),
+                              skip = 1,  # skip version string
+                              data.table = FALSE)
+```
+
+```{r}
+sample_id_list$mutect2 <- unique(mutect2$Tumor_Sample_Barcode)
+sample_id_list$strelka2 <- unique(strelka2$Tumor_Sample_Barcode)
+```
+
+## SV 
+
+The Manta SV file is in [annotated Manta TSV format](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/format/manta-tsv-header.md).
+
+```{r}
+manta <- readr::read_tsv(file.path(data_dir, manta_file))
+```
+
+```{r}
+sample_id_list$manta <- unique(manta$Kids.First.Biospecimen.ID.Tumor)
+```
+
+```{r}
+#TODO: LUMPY SV
+```
+
+## Select samples and subset files
+
+### Map to participant identifiers
+
+These sample identifiers are all biospecimen identifiers.
+The tools that use different experiment strategies will have no overlapping
+biospecimen identifiers.
+Thus, we'll need to map _biospecimen_ identifiers to _participant_ identifiers.
+We'll use the histologies file to do this.
+
+```{r}
+histologies_df <- readr::read_tsv(file.path(data_dir, histologies_file))
+```
+
+```{r}
+participant_id_list <- lapply(
+  sample_id_list,
+  function(sample_ids) {
+    histologies_df %>%
+      dplyr::filter(Kids_First_Biospecimen_ID %in% sample_ids) %>% 
+      dplyr::pull(Kids_First_Participant_ID)
+  }
+)
+```
+
+Now we can take the intersection of elements of `participant_id_list` to
+identify participant identifiers represented in each of the files we're trying
+to subset.
+
+```{r}
+participants_in_all <- Reduce(intersect, participant_id_list)
+```
+
+### Select samples
+
+We'll select 5 participant identifiers for the purposes of producing subset
+files (for continuous integration).
+
+```{r}
+# set seed for reproducibility
+set.seed(12345)
+selected_participant_ids <- sample(participants_in_all, 5)
+```
+
+Again, each of the tools use biospecimen identifiers. 
+We'll have to map back to those to accomplish the subsetting task.
+
+```{r}
+selected_biospecimen_df <- histologies_df %>%
+  dplyr::filter(Kids_First_Participant_ID %in% selected_participant_ids)
+```
+
+`selected_biospecimen_df` is our subset of `pbta-histologies.tsv`.
+
+```{r}
+readr::write_tsv(selected_biospecimen_df,
+                 file.path(testing_data_dir, "pbta-histologies.tsv"))
+```
+
+To keep things less verbose below, we'll keep the biospecimen IDs as a vector.
+
+```{r}
+selected_biospecimen_ids <- selected_biospecimen_df %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+```
+
+### Subset files
```

Comment:
Are there samples that are not in all the files? Should we have some sort of check or error handling to let the person know that a particular sample they are trying to obtain in the subset doesn't exist in a particular dataset?

Reply:
This is beyond the scope of this pull request (and certainly beyond the scope of my knowledge of these data). My intention with this PR is to generate files that can be used as test data for continuous integration, so in this specific case, it does not matter.

More generally, I would expect folks to find the 'missing' samples organically through their own analyses and as we get the first couple ones merged in.

Reply:
As an FYI, I had added the `experimental_strategy` column in the `pbta-histologies.tsv` file, so all samples with RNA-Seq will have fusions, expression, etc, those with WGS/Panel/WES should have SNVs, those with WGS should have CN and SVs.